### DISCUSSION TITLE
Return `matches` information even if there is no `attributesToRetrieve`

### DISCUSSION THREAD

**Discussion by curquiza** [meilisearch/product#280] [Feedback & Feature Proposal]
Following the use case of this user, this might be interesting to consider returning the `matches` information, no matter how are the `attributesToRetrieve`.

Following https://github.com/meilisearch/MeiliSearch/issues/1825

We should also consider the users that definitely don't want the information related to an attribute if they don't put it in `attributesToRetrieve`.

**Comment by gmourier** [comment]
Thanks, @curquiza!

**Comment by giray123** [comment]
I think my problem is also related to this. Here is my use case:

I have a very long text attribute. I only want to retrieve a cropped version of this attribute if there is a match with the searched text. Because this very long text slows down the query response times. I have tried many different configurations (one example is below) but there seems to be not an easy solution. This also relates to this issue but can be a different suggestion issue as well.

```
{
  attributesToRetrieve: ['shortText'],
  attributesToHighlight: ['shortText', 'longText'],
  attributesToCrop = ['longText']
  cropLength = 10
  matches = true
}
```
